kbase-extension/static/kbase/js/widgets/appWidgets2/input/dynamicDropdownInput.js

```diff
@@ -332,53 +317,77 @@ define([
 
         /**
          * Formats the display of an object in the dropdown.
-         id: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
-         isFolder: false
-         mtime: 1508441424000
-         name: "i_am_a_file.txt"
-         path: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
-         size: 0
-         text: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
+         * id: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
+         * isFolder: false
+         * mtime: 1508441424000
+         * name: "i_am_a_file.txt"
+         * path: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
+         * size: 0
+         * text: "data/bulk/wjriehl/subfolder/i_am_a_file.txt"
          */
-        function formatObjectDisplay(ret_obj) {
-            if (!ret_obj.id) {
-                return '';
+        function formatObjectDisplay(retObj) {
+            if (!retObj.id) {
+                return retObj.text || '';
```

Comment:
This fixes the `Loading...` bit. Apparently every `Option` element in Select2 gets pushed through the formatter, including those that Select2 inserts itself. Seems a bit of a bug on their side, IMO, but this fixes it. Anything that gets used from a service to be formatted for display is expected to be in the form of an object with an `id` field anyway.